include/gz/rendering/Material.hh

```diff
@@ -220,8 +221,16 @@ namespace gz
       public: virtual std::string Texture() const = 0;
 
       /// \brief Set the material texture
-      /// \param[in] _texture URI of the new texture file
-      public: virtual void SetTexture(const std::string &_texture) = 0;
+      /// \param[in] _texture URI of the new texture file or identifying name
+      /// if _img is set
+      /// \param[in] _img Image data
+      public: virtual void SetTexture(const std::string &_texture,
+        const std::shared_ptr<common::Image> &_img = nullptr) = 0;
```

Comment:
I think this change needs a tick-tock, otherwise downstream projects who currently implement `SetTexture` with the old signature on Fortress won't compile against Garden without making changes to the code.

How about keeping the old function and making this one like this:

```cpp
      public: virtual void SetTexture(const std::string &_texture,
        const std::shared_ptr<common::Image> &_img)
     {
       this->SetTexture(_texture);
     }
```


i.e. not pure virtual and without a default value.

Reply:
I thought that adding a default argument would be OK, the function should be able to be called with either:

```
// Old behavior, uses default parameter of _img = nullptr;
SetTexture(name);
// New behavior, assigns an image
SetTexture(name, image);
```

Reply:
It's ok for whoever calls the function, but classes inheriting from `Material` will need to update the function signature

Reply:
I implemented a similar suggestion for light maps in https://github.com/gazebosim/gz-rendering/pull/642/commits/a9be29c93ea27115551fa02711cc8a8e4d7b0f8c, if it looks good I'll make the same change for all the other functions as well.
Should I add deprecation warnings to the pure virtual methods so we can tock in Gazebo H?

Reply:
Cool! I'm ok adding a deprecation warning. Let's just make that not-pure-virtual anymore, so subclasses on Garden can opt out of it.

Reply:
Done, updated all of them! https://github.com/gazebosim/gz-rendering/pull/642/commits/19c542727ec168cdd1e8521bf05415f8b3c34f1b I had to change the implementation to a noop instead of calling the deprecated function otherwise there would be a lot of deprecation warnings printed everytime someone imported `Material.hh`